libcaf_core/caf/scheduler.hpp

```diff
@@ -4,6 +4,85 @@
 
 #pragma once
 
-#include "caf/scheduler/abstract_coordinator.hpp"
-#include "caf/scheduler/coordinator.hpp"
-#include "caf/scheduler/worker.hpp"
+#include "caf/actor_clock.hpp"
+#include "caf/actor_system.hpp"
+#include "caf/detail/actor_local_printer.hpp"
+#include "caf/detail/core_export.hpp"
+#include "caf/fwd.hpp"
+
+#include <cstddef>
+
+namespace caf {
+
+/// A coordinator creates the workers, manages delayed sends and
+/// the central printer instance for {@link aout}. It also forwards
+/// sends from detached workers or non-actor threads to randomly
+/// chosen workers.
+class CAF_CORE_EXPORT scheduler : public actor_system::module {
+public:
+  // -- factory functions ------------------------------------------------------
+  static scheduler* make_work_stealing(actor_system& sys);
+
+  static scheduler* make_work_sharing(actor_system& sys);
+
+  // -- scheduler interface ----------------------------------------------------
+
+  /// Puts `what` into the queue of a randomly chosen worker.
+  virtual void enqueue(resumable* what) = 0;
+
+  virtual actor_clock& clock() noexcept = 0;
+
+  // -- implementation of actor_system::module ---------------------------------
+
+  void start() override;
+
+  void init(actor_system_config& cfg) override;
+
+  id_t id() const override;
+
+  void* subtype_ptr() override;
+
+  // -- utility functions ------------------------------------------------------
+
+  static void cleanup_and_release(resumable*);
+
+  virtual detail::actor_local_printer_ptr printer_for(local_actor* self);
+
+  // -- properties -------------------------------------------------------------
+
+  static size_t default_thread_count() noexcept {
+    return std::max(std::thread::hardware_concurrency(), 4u);
+  }
```

Comment:
I think this should move to the `.cpp` file, even if it's a small function. If we ever change the default around, we want linked applications to use the new default.

Reply:
This was in a cpp but I inlined it thinking it's a short function. Moving back.